x-pack/plugins/apm/server/lib/errors/get_error_rate.ts

```diff
@@ -4,106 +4,110 @@
 import {
   Setup,
   SetupTimeRange,
   SetupUIFilters,
 } from '../helpers/setup_request';
-import { rangeFilter } from '../../../common/utils/range_filter';
 
 export async function getErrorRate({
   serviceName,
-  groupId,
   setup,
 }: {
   serviceName: string;
-  groupId?: string;
   setup: Setup & SetupTimeRange & SetupUIFilters;
 }) {
   const { start, end, uiFiltersES, client, indices } = setup;
 
   const filter = [
     { term: { [SERVICE_NAME]: serviceName } },
+    { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
     { range: rangeFilter(start, end) },
     ...uiFiltersES,
   ];
 
-  const aggs = {
-    response_times: {
+  const must = [{ exists: { field: HTTP_RESPONSE_STATUS_CODE } }];
+
+  const dateHistogramAggs = {
+    histogram: {
       date_histogram: getMetricsDateHistogramParams(start, end),
     },
   };
 
-  const getTransactionBucketAggregation = async () => {
-    const resp = await client.search({
+  const getTransactionsCount = async () => {
+    const transactionsCountParams = {
       index: indices['apm_oss.transactionIndices'],
       body: {
         size: 0,
-        query: {
-          bool: {
-            filter: [
-              ...filter,
-              { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
-            ],
-          },
-        },
-        aggs,
+        query: { bool: { must, filter } },
+        aggs: dateHistogramAggs,
       },
-    });
+    };
+
+    const resp = await client.search(transactionsCountParams);
+    const transactionsCountByTimestamp: Record<number, number> = {};
+    if (resp.aggregations) {
+      resp.aggregations.histogram.buckets.forEach(
+        (bucket) =>
+          (transactionsCountByTimestamp[bucket.key] = bucket.doc_count)
+      );
+    }
     return {
-      totalHits: resp.hits.total.value,
-      responseTimeBuckets: resp.aggregations?.response_times.buckets,
+      transactionsCountByTimestamp,
+      noHits: resp.hits.total.value === 0,
     };
   };
-  const getErrorBucketAggregation = async () => {
-    const groupIdFilter = groupId
-      ? [{ term: { [ERROR_GROUP_ID]: groupId } }]
-      : [];
-    const resp = await client.search({
-      index: indices['apm_oss.errorIndices'],
+
+  const getErroneousTransactionsCount = async () => {
+    const erroneousTransactionsCountParams = {
+      index: indices['apm_oss.transactionIndices'],
       body: {
         size: 0,
         query: {
           bool: {
+            must,
             filter: [
               ...filter,
-              ...groupIdFilter,
-              { term: { [PROCESSOR_EVENT]: ProcessorEvent.error } },
+              {
+                range: {
+                  [HTTP_RESPONSE_STATUS_CODE]: {
+                    gte: 400, // everything equals or above 400 should be treated as an error
+                  },
+                },
+              },
             ],
           },
         },
-        aggs,
+        aggs: dateHistogramAggs,
       },
-    });
```

Comment:
You can reduce the two queries to a single one! It will simplify this a whole lot :)

Small demo I did:
```json
GET apm-8.0.0-transaction-000001/_search
{
  "size": 0,
  "query": {
    "bool": {
      "filter": [
        {
          "exists": {
            "field": "http.response.status_code"
          }
        }
      ]
    }
  },
  "aggs": {
    "total_transactions": {
      "date_histogram": {
        "field": "@timestamp",
        "interval": "day"
      },
      "aggs": {
        "errornous_transactions": {
          "filter": {
            "range": {
              "http.response.status_code": {
                "gte": 300
              }
            }
          }
        }
      }
    }
  }
}
```

**Response**
```json
{
  "aggregations": {
    "total_transactions": {
      "buckets": [
        {
          "key_as_string": "2020-03-30T00:00:00.000Z",
          "key": 1585526400000,
          "doc_count": 13028,
          "errornous_transactions": {
            "doc_count": 1132
          }
        },
        {
          "key_as_string": "2020-03-31T00:00:00.000Z",
          "key": 1585612800000,
          "doc_count": 39551,
          "errornous_transactions": {
            "doc_count": 3328
          }
        }
      ]
    }
  }
}

```